web/src/components/tasks/CannabisPriorityStatus.tsx

```diff
@@ -0,0 +1,342 @@
+  ],
+  socialEquity: [
+    "cannabis-economically-disadvantaged-social-equity",
+    "cannabis-criminal-offense-social-equity",
+  ],
+  noneOfTheAbove: "cannabis-priority-status-none",
+};
+
+export const CannabisPriorityStatus = (props: Props) => {
+  const { userData, update } = useUserData();
+  const taskFromRoadmap = useTaskFromRoadmap(props.task.id);
+  const unlockedByTask = getUnlockedBy(taskFromRoadmap, userData);
+  const [successToastIsOpen, setSuccessToastIsOpen] = useState(false);
+
+  const [displayFirstTab, setDisplayFirstTab] = useState(true);
+  const [displayNextTabButton, setDisplayNextTabButton] = useState(false);
+
+  const [displayMWPriorityType, setDisplayMWPriorityType] = useState(false);
+  const [displayVeteranPriorityType, setDisplayVeteranPriorityType] = useState(false);
+  const [displayImpactZonePriorityType, setDisplayImpactZonePriorityType] = useState(false);
+  const [displaySocialEquityPriorityType, setDisplaySocialEquityPriorityType] = useState(false);
+  const [displayNoPriorityType, setDisplayNoPriorityType] = useState(false);
+
+  const priorityTypes = useMemo(() => {
+    return [
+      ...priorityTypesObj.minorityOrWomen,
+      ...priorityTypesObj.veteran,
+      ...priorityTypesObj.impactZone,
+      ...priorityTypesObj.socialEquity,
+    ];
+  }, []);
+
+  useEffect(() => {
+    if (userData && userData.taskItemChecklist && displayFirstTab) {
+      const priorityTypeSelected = priorityTypes.some((key) => userData.taskItemChecklist[key] === true);
+
+      if (priorityTypeSelected || userData.taskItemChecklist[priorityTypesObj.noneOfTheAbove]) {
+        setDisplayNextTabButton(true);
+      } else setDisplayNextTabButton(false);
+
+      if (priorityTypeSelected && userData.taskItemChecklist[priorityTypesObj.noneOfTheAbove]) {
+        update({
+          ...userData,
+          taskItemChecklist: {
+            ...userData.taskItemChecklist,
+            [priorityTypesObj.noneOfTheAbove]: false,
+          },
+        });
+      }
+    }
+
+    if (userData && userData.taskItemChecklist && !displayFirstTab) {
+      const minorityOrWomenPriorityTypeSelected = priorityTypesObj.minorityOrWomen.some((key) => {
+        if (key in userData.taskItemChecklist) return userData.taskItemChecklist[key] === true;
+        else return false;
+      });
+      const veteranPriorityTypeSelected = priorityTypesObj.veteran.some((key) => {
+        if (key in userData.taskItemChecklist) return userData.taskItemChecklist[key] === true;
+        else return false;
+      });
+
+      const impactZonePriorityTypeSelected = priorityTypesObj.impactZone.some((key) => {
+        if (key in userData.taskItemChecklist) return userData.taskItemChecklist[key] === true;
+        else return false;
+      });
+
+      const socialEquityPriorityTypeSelected = priorityTypesObj.socialEquity.some((key) => {
+        if (key in userData.taskItemChecklist) return userData.taskItemChecklist[key] === true;
+        else return false;
```

Comment:
these could all be refactored to `return userData.taskItemChecklist[key] === true` right? because if the key isn't in `userData.taskItemChecklist` then `userData.taskItemChecklist[key]` will return undefined